kubernetes-addons/argocd/main.tf

```diff
@@ -73,5 +73,42 @@ resource "helm_release" "argocd" {
       value = each_item.value.value
     }
   }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# ArgoCD App of Apps Bootstrapping
+# ---------------------------------------------------------------------------------------------------------------------
 
+resource "kubernetes_manifest" "argocd_application" {
```

Comment:
@vara-bonthu lets discuss

Reply:
Just seen the limitation on this from Hasicorp blog. We will find a workaround. 
`Access to the Kubernetes API is required at plan time` . This means that a Kubernetes cluster must be created and available in a separate Terraform apply operation before you can apply a kubernetes_manifest resource. This is because the provider has to fetch the OpenAPI specification from the Kubernetes API to generate the type information required by Terraform.